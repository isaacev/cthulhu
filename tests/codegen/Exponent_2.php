<?php

namespace Exponent_2 {
  function main() {
    \pow(2.0, 0.5);
  }
}

namespace {
  \Exponent_2\main(null);
}
