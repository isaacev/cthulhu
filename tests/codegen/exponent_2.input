#[entry]
fn main() -> () {
  let n: ::Kernel::Float = 2 ^ 0.5;
}
