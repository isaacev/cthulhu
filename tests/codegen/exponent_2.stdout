<?php

namespace Kernel\Types {
  abstract class Maybe {}

  class Just extends \Kernel\Types\Maybe {
    function __construct($a) {
      $this->{0} = $a;
    }
  }

  class None extends \Kernel\Types\Maybe {}
}

namespace exponent_2 {
  // #[entry]
  function main() {
    $n = 2 ** 0.5;
  }
}

namespace {
  \exponent_2\main();
}
